web/src/app/modules/overview/services/navigator/navigator.service.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2020 the Octant contributors. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+
+export interface NavigatorStatus {
+  history: string[];
+  index: number;
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class NavigatorService {
+  private internalHistory: string[] = [];
+  private internalIndex = -1;
+
+  private source = new BehaviorSubject<NavigatorStatus>(null);
+  status = this.source.asObservable();
+
+  constructor() {}
+
+  addHistory(path: string) {
+    console.log(
+      `tail: ${
+        this.internalHistory[this.internalHistory.length - 1]
+      }; path: ${path}`
```

Comment:
URLs with a query param (such as clicking on tabs) will be encoded and causes navigator to go to an invalid path. A filter might be needed.